### DISCUSSION TITLE
Edit Bard Content with Antlers/modifiers

### DISCUSSION THREAD

**Discussion by eliasu** [statamic/cms#4527] [Q&A] [answered]
**Hi there,** 

I have a Bard-field inside a collection named "quote-text". Bard is configured, that you can only use **bold** and _italic_.
Is it possible to add **"»"** at the beginning and **"«"** at the end of this Bard-text via Antlers/modifiers (so that we don't change Bard's structure in YAML)?

**Any idea?** 
Thanks in advance.

**Comment by duncanmcclean** [comment, marked as answer]
Does your Bard field have any sets?

**Reply by eliasu** [reply]
no.

**Reply by duncanmcclean** [reply]
Just written a little modifier that'll do it for you. Create a file `app/Modifiers/InQuotes.php`, with these contents:

```php
<?php

namespace App\Modifiers;

use Statamic\Modifiers\Modifier;

class InQuotes extends Modifier
{
    public function index($value, $params, $context)
    {
        $value = $this->strReplaceFirst('<p>', '<p>» ', $value);
        $value = $this->strReplaceLast('</p>', '«</p>', $value);

        return $value;
    }

    protected function strReplaceFirst($from, $to, $content)
    {
        $from = '/' . preg_quote($from, '/') . '/';

        return preg_replace($from, $to, $content, 1);
    }

    protected function strReplaceLast($search, $replace, $str)
    {
        if (($pos = strrpos($str, $search)) !== false) {
            $searchLength = strlen($search);
            $str = substr_replace($str, $replace, $pos, $searchLength);
        }

        return $str;
    }
}
```

Then you can use your Bard field like so:

```antlers
{{ bard_field | in_quotes }}
```

I'm sure the code can be cleaned up a little if you want but that's what I could whip up quickly. Hopefully it does the job!

**Reply by jasonvarga** [reply]
This might also work, for single paragraphs.
```
{{ if bard_text }}
  »{{ bard_text|striptags }}«
{{ /if }}
```

**Reply by eliasu** [reply]
@duncanmcclean Thanks a lot. That's exactly what I was looking for. It works! 👌

@jasonvarga That works too, but like you said, only for single paragraphs. To "protect" clients from layout problems, the answer above is more failproof 😉
